src/main/scala/com/swoop/spark/records/package.scala

```diff
@@ -32,14 +32,14 @@ package object records {
     *
     * @see [[RootCauseAnalysisOps]]
     */
-  implicit class RecordsRichDataset[A <: Product : Encoder, Rec](val underlying: Dataset[Rec])
+  implicit class RecordsRichDataset[A <: Product, Rec](val underlying: Dataset[Rec])
     (implicit ev: Rec <:< Record[A, _]) extends Serializable {
 
     def errorRecords(implicit env: RecordEnvironment): Dataset[Rec] =
       env.errorRecords(underlying)
 
     def allIssues(implicit env: RecordEnvironment): DataFrame =
-      new RootCauseAnalysis(underlying.filter(_.issues.isDefined).toDF(), env).issues
+      new RootCauseAnalysis(underlying.filter((r: Rec) => r.issues.isDefined).toDF(), env).issues
```

Comment:
Do you get a compiler error without the refactoring?

Reply:
@ssimeonov Yes I did get a compiler error asking me to implicitly define the type. Seems that the new overloaded methods were introduced in last Spark version so this one became polymorphic, so we have to be explicit on this.